File: backend/app/models/dso_visibility.py
"""Pydantic schemas for the DSO visibility API."""

from pydantic import BaseModel, Field

from app.models.astronomy import VisibilityWindow


class DsoVisibilityRequest(BaseModel):
    latitude: float = Field(..., ge=-90, le=90)
    longitude: float = Field(..., ge=-180, le=180)
    timezone: str = Field(..., min_length=1)
    dates: list[str] = Field(..., min_length=1, max_length=7)


class SiteSkyConditions(BaseModel):
    bortle: int
    sqm: float
    limiting_magnitude: float
    source: str


class DsoVisibilityRow(BaseModel):
    id: str
    name: str
    common_name: str | None = None
    object_type: str
    visible: bool
    windows_astronomical: list[VisibilityWindow] = Field(default_factory=list)
    peak_altitude_deg: float | None = None
    peak_at: str | None = None
    magnitude: float | None = None
    contrast: float
    visibility_score: float


class DsoDayVisibility(BaseModel):
    date: str
    objects: list[DsoVisibilityRow]


class DsoVisibilityResponse(BaseModel):
    site_sky: SiteSkyConditions
    dso_visibility: list[DsoDayVisibility]

File: frontend/src/types/dso-visibility.ts
import type { VisibilityWindow } from './astronomy'

export interface SiteSkyConditions {
  bortle: number
  sqm: number
  limiting_magnitude: number
  source: string
}

export interface DsoVisibilityRow {
  id: string
  name: string
  common_name: string | null
  messier: number | null
  object_type: string
  visible: boolean
  windows_astronomical: VisibilityWindow[]
  peak_altitude_deg: number | null
  peak_at: string | null
  magnitude: number | null
  contrast: number
  visibility_score: number
}

export interface DsoDayVisibility {
  date: string
  objects: DsoVisibilityRow[]
}

export interface DsoVisibilityRequest {
  latitude: number
  longitude: number
  timezone: string
  dates: string[]
}

export interface DsoVisibilityResponse {
  site_sky: SiteSkyConditions
  dso_visibility: DsoDayVisibility[]
}
